### PR TITLE
adding the PandasXml Reader and Writer Class

### PR DESCRIPTION
**Pull request by JoJo10Smith**
This code is for ticket #352 where I will adding Pandas XML functionality to read and write to that file format. The methods used and kwargs are described: 

- Reader: https://pandas.pydata.org/docs/reference/api/pandas.read_xml.html
- Writer: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_xml.html

## Changes

The changes are to the following files:

- `test_pandas_extensions.py `==> I added a few tests that I will run to test the functionality of both new methods
- `pandas_extensions.py `==> added the classes for reading and writing to xml
- `test_load_from_data.xml `==> a new file with some data in XML format to test

## How I tested this

I have read the dev guidelines and contribution notes and installed to test locally. Unfortunately, I am still struggling with getting Circleci to run on my branch, and will make another attempt to test this at a later stage.

## Notes
This commit is still a work-in-progress (WIP) and has not been tested. 

When I run `circleci local execute --job=JOB_NAME` where I change `JOB_NAME` to be `run tests` according to the `confing.yml `or even `build-py37` I get the error: `Error: unknown flag: --job
`

## Checklist
- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by skrawcz** [comment]
For the pre-commit ones, please install the pre-commit hook see [this doc](https://github.com/DAGWorks-Inc/hamilton/blob/main/developer_setup.md#set-up-your-local-dev-environment).

> When I run circleci local execute --job=JOB_NAME where I change JOB_NAME to be run tests according to the confing.yml or even build-py37 I get the error: Error: unknown flag: --job 

Ah we might need to update our documentation. I think `circleci` updated their CLI. and now you don't need the `--job=` argument.

**Comment by JoJo10Smith** [comment]
I see, I'll run the tests tomorrow after your recommendation. I can also change the docs once I get it working as well. Thank @skrawcz

**Comment by JoJo10Smith** [comment]
I needed to change the assertion test I was running since I had incorrectly included the column heading as part of the count for when I was conducting the test. As shown in the image below there are 4 data columns, not 5.

Screenshot from 2023-09-16 11-44-32

**Comment by JoJo10Smith** [comment]
@skrawcz this ticket is ready for review. I saw that in issue #346 the only files that were changed were the pandas_extension.py file and the test_pandas_extensions.py file. I'm not sure where you would like me to document the added functionality to meet the last 3 requirements before this code can be merged.

Thank You
Jordan

**Comment by skrawcz** [comment]
> @skrawcz this ticket is ready for review. I saw that in issue #346 the only files that were changed were the pandas_extension.py file and the test_pandas_extensions.py file. I'm not sure where you would like me to document the added functionality to meet the last 3 requirements before this code can be merged.
> 

Sure, take a look at https://github.com/DAGWorks-Inc/hamilton/commit/abd96448d7acc10c3ff79da3cfa6d9eef19c5288. Basically it adds to the https://github.com/DAGWorks-Inc/hamilton/tree/main/examples/pandas/materialization example.

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+lxml
```
Comment (skrawcz) [inline requirements.txt:1]: Can you move this to `requirements-test.txt` please?  The reason being that we want to keep Hamilton's main dependencies nice and light.

Adding to `requirements-test.txt` will ensure the tests have the right dependencies.

**Review by skrawcz — CHANGES_REQUESTED** [review]
Just want to move the `lxml` dependency, and note that it's required for this materializer to work. Otherwise looking good!

File: examples/pandas/materialization/notebook.ipynb
Comment (JoJo10Smith) [inline examples/pandas/materialization/notebook.ipynb:1]: Just need additional help running the `.ipynb file`. When I open it in VSCode I can run everything but get an error when trying to import the hamilton module. I need to edit this file to add the new XML materialization example.

Screenshot from 2023-09-16 17-28-55

File: requirements.txt (hunk 1 above)
Comment (JoJo10Smith) [reply requirements.txt]: Changed the location of the module to be in the `requirements-test.txt` file. Marking this conversation as resolved.

File: examples/pandas/materialization/notebook.ipynb
Comment (skrawcz) [reply examples/pandas/materialization/notebook.ipynb:1]: you'll need hamilton on the python path to however you started the notebook. How are you starting the notebook?

File: hamilton/plugins/pandas_extensions.py [hunk 2]
```diff
@@ -383,6 +383,166 @@ def name(cls) -> str:
...
+@dataclasses.dataclass
+class PandasXmlReader(DataLoader):
+    """Class for loading/reading xml files with Pandas.
+        Maps to https://pandas.pydata.org/docs/reference/api/pandas.read_xml.html
+    <<<<<<< HEAD
+
+        Requires `lxml`. See https://pandas.pydata.org/docs/getting_started/install.html#xml
+    =======
+        Requires `lxml`. See https://pandas.pydata.org/docs/getting_started/install.html#xml.
+    >>>>>>> c59421c8cf53af8ea5e8c092ce96621de8643ffc
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:395]: can you fix this part up please? :)

File: hamilton/plugins/pandas_extensions.py [hunk 3]
```diff
@@ -383,6 +383,166 @@ def name(cls) -> str:
...
+
+@dataclasses.dataclass
+class PandasXmlWriter(DataSaver):
+    """Class specifically to handle saving xml files/buffers with Pandas.
+        Should map to https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_xml.html
+    <<<<<<< HEAD
+
+    =======
+    >>>>>>> c59421c8cf53af8ea5e8c092ce96621de8643ffc
+        Requires `lxml`. See https://pandas.pydata.org/docs/getting_started/install.html#xml.
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:478]: ditto

**Pull request closed without merging**